lms/djangoapps/course_api/v0/views.py

```diff
@@ -0,0 +1,451 @@
+        data['org'] = content_descriptor.location.org
+        data['run'] = content_descriptor.location.run
+
+    # Other things we do only if the content object is not a course
+    else:
+        content_id = unicode(content_descriptor.location)
+        # Need to use the CourseKey here, which will possibly result in a different (but valid)
+        # URI due to the change in key formats during the "opaque keys" transition
+        content_uri = request.build_absolute_uri(reverse('course_api:v0:content:detail',
+                                                         kwargs={'course_id': unicode(course_key),
+                                                                 'content_id': content_id}))
+
+    data['id'] = unicode(content_id)
+    data['uri'] = content_uri
+
+    # Include any additional fields requested by the caller
+    include_fields = request.QUERY_PARAMS.get('include_fields', None)
+    if include_fields:
+        include_fields = include_fields.split(',')
+        for field in include_fields:
+            data[field] = getattr(content_descriptor, field, None)
+
+    return data
+
+
+def _serialize_content_children(request, course_key, children):
+    """
+    Loads the specified content child data into the response dict
+    This should probably evolve to use DRF serializers
+    """
+    data = []
+    if children:
+        for child in children:
+            child_data = _serialize_content(
+                request,
+                course_key,
+                child
+            )
+            data.append(child_data)
+    return data
+
+
+def _serialize_content_with_children(request, course_key, descriptor, depth):  # pylint: disable=invalid-name
+    """
+    Serializes course content and then dives into the content tree,
+    serializing each child module until specified depth limit is hit
+    """
+    data = _serialize_content(
+        request,
+        course_key,
+        descriptor
+    )
+
+    if depth > 0:
+        data['children'] = []
+        for child in descriptor.get_children():
+            data['children'].append(_serialize_content_with_children(request, course_key, child, depth - 1))
+    return data
+
+
+def _get_course_data(request, course_key, course_descriptor, depth=0):
+    """
+    creates a dict of course attributes
+    """
+
+    if depth > 0:
+        data = _serialize_content_with_children(
+            request,
+            course_key,
+            course_descriptor,  # Primer for recursive function
+            depth
+        )
+        data['content'] = data['children']
+        data.pop('children')
+    else:
+        data = _serialize_content(
+            request,
+            course_key,
+            course_descriptor
+        )
+
+    data['image_url'] = ''
+    if getattr(course_descriptor, 'course_image'):
+        data['image_url'] = course_image_url(course_descriptor)
+
+    return data
+
+
+class CourseContentList(APIViewWithPermissions):
```

Comment:
It seems odd to have an endpoint that returns a list of children of the course instead of just the tree with the course as the root.